src/Sheet.php

```diff
@@ -253,8 +259,29 @@ public function import($import, int $startRow = 1)
 
         if ($import instanceof OnEachRow) {
             $headingRow = HeadingRowExtractor::extract($this->worksheet, $import);
+
+            $rows = [];
+
             foreach ($this->worksheet->getRowIterator()->resetStart($startRow ?? 1) as $row) {
-                $import->onRow(new Row($row, $headingRow));
+                $rows[] = new Row($row, $headingRow);
```

Comment:
As said in previous review, this will load all rows in memory, defeating the purpose of the concern. The validation has to be performed inside the loop and rows shouldn't be kept in memory.